### DISCUSSION TITLE
General feedback

### DISCUSSION THREAD

**Discussion by CrawlerCode** [CrawlerCode/redmine-time-tracking#1] [General]
### Feel free to give me feedback

- If you found a bug, please open a [issue](https://github.com/CrawlerCode/redmine-time-tracking/issues/new?template=bug_report.yml)
- If you have a feature request or ideas. Do it [here](https://github.com/CrawlerCode/redmine-time-tracking/discussions/new?category=feature-request-ideas)

_Thanks :)_

**Comment by ASM-Development** [comment]
Rounding is required: https://yadi.sk/i/kz05camtJr4o8g

**Reply by CrawlerCode** [reply]
Thank you for reporting this issue.

I fixed the issue and uploaded a new version (`v1.11.1`) to the [Chrome Web Store](https://chrome.google.com/webstore/detail/redmine-time-tracking/ldcanhhkffokndenejhafhlkapflgcjg), it should be available tomorrow.

**Comment by ASM-Development** [comment]
1. I have added a translation into Russian.
Please add it to the project.
[ru.zip](https://github.com/CrawlerCode/redmine-time-tracking/files/12593213/ru.zip)

2. Incorrect labels.
Incorrect: https://yadi.sk/i/aqGR39mIF8_lVg, correct: Click to pause timer
Incorrect: https://yadi.sk/i/hvHeWmTm4geIsw, correct: Click to start timer

**Reply by CrawlerCode** [reply]
Thank you for contribution and reporting issues.

I have added Russian as a new language and fixed the issue #3.
_For the future, please use the isses feature of Github to report new issues or feature requests. Thanks :)_

New version (`v1.11.2`) has been released and uploaded.

**Comment by ASM-Development** [comment]
Here https://chrome.google.com/webstore/detail/redmine-time-tracking/ldcanhhkffokndenejhafhlkapflgcjg the version is still 1.11.1.
Tell me when it will be updated to the new v1.11.2 ?

**Reply by CrawlerCode** [reply]
Google reviews every new version, this can take up to 30 days. But in most cases it takes only 1 day.
You can find more information [here](https://developer.chrome.com/docs/webstore/review-process)

**Comment by dkochc** [comment]
Hello,
Is it possible to pin tasks whose time is currently being tracked to the top of the list? I have a lot of tasks in my list and when I work on several tasks at the same time, I either have to scroll a lot or use the search function to find the relevant entries.

Kind regards

**Reply by felipoantonoff** [reply]
I also find this feature interesting, those who have many tasks and projects may get lost which ones have the Timer active or paused.

**Reply by CrawlerCode** [reply]
Thank you for your ideas.


I have added the feature that tracked issues can now be pinned at the top. 
This feature is disabled by default and you must enable it in the settings.
![image](https://github.com/CrawlerCode/redmine-time-tracking/assets/41094392/6b9024c4-e8d5-4043-a804-8aff40a5c3e2)

It will be available in the new version (`v1.16.0`)

**Comment by felipoantonoff** [comment]
Hello @CrawlerCode , first of all thanks for the great extension, it will help a lot to control the time of tasks using Redmine.

**I tested it quickly, but I'll point out possible suggestions for improvements:**
- When you click on Stop, you shouldn't waste time, perhaps a warning will appear informing you if you want to save the time in Redmine, because if you click wrong you could lose all the time for the task or maybe just confirm if you really want to waste the time, it will stay very close to start/pause and save
- Highlight all the tasks that have the active Timer running and then the paused ones even before the Pins, maybe even have a Tab just for the Timers in progress, something like Run or take advantage of the Time tab
- In Redmine, the task page could have a shortcut to start it, similar to what Toggl does, so we don't need to search for the task in the extension but instead start or pause it directly from the task in Redmine and there could also be a shortcut to save When Spent clicks on it, it opens the extension box to save, for example
Imageexample:
![image](https://github.com/CrawlerCode/redmine-time-tracking/assets/3225997/992fec6b-f186-420c-94e4-a26bf0237b55)
- You can automatically save Spent Time, perhaps a quick button that already saves directly without a note and marking the default activity

**Reply by CrawlerCode** [reply]
Thank you for your ideas.

- I have implemented a confirmation dialog for resetting a timer. And I renamed "stop" to "reset", that should be a better wording.
- Tracked issues can now be pinned at the top. (see [above](https://github.com/CrawlerCode/redmine-time-tracking/discussions/1#discussioncomment-8833290))
- Regarding the "quick button", I need to get more opinions and possibly rethink the reset button.

The other suggestions need a little more time. I will probably implement them in the future.

**Comment by varnak-github** [comment]
Hello,  @CrawlerCode 
Thank you very much for your great plugin.

I have encountered an issue with the "Extended Search" option (which allows searching for issues that are not assigned to you). The main concern is that I currently do not have the ability to select issues that are not assigned to me, which seems incorrect. I'm not entirely sure if I understood the purpose of the "Extended Search" option correctly, but I believe it should allow me to find any issue within my Redmine project. Additionally, I should be able to track my time on any issue, even if it is not assigned to me.

Could you please clarify if my understanding is correct? Is this behavior a bug, or is it intended to work this way?

Thank you for your assistance.
![Screen_2024-08-22_13-10](https://github.com/user-attachments/assets/711faeb3-4431-4b3d-b1c0-4095ea5289b4)

**Reply by CrawlerCode** [reply]
First of all, there is the normal search (search in your own issues). The search can be opened by pressing CTRL + K or CTRL + F.
If the “extended search” feature is activated, you coud search issues that is not assigned to you (displayed after the normal search). At this point you can start the timer and do anything you want with this issue, as with all other issues.
![image](https://github.com/user-attachments/assets/457d4e68-ca79-4f84-b8ed-157b820045e7)

I don't know exactly what you mean by “ability to select issues”. If what i have described above does not work, it may be a bug. Then please describe this in detail and tell me the version of your redmine server.

**Reply by ghost** [reply]
Hello,
first thank you for this plugin. I wish I would have found that earlier...
Currently I am evaluating it for installing in our company by the end of the month.
I have three feature requests.
 
1. Could you add a "Search always visible" setting?
In our company we have many tickets and need the search all the time. 

2. Pin currently open ticket to top.
What I mean is when the the redmine ticket page (http://redmine_srv.com/issues/1878) is the current tab and I click on the extension, the ticket with the issue number should be pinned to the top.

```
chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
  var activeTab = tabs[0];
  var url = activeTab.url;

  // Extract the issue number
  var urlParts = url.split("/");
  var issueNumber = urlParts[urlParts.length - 1]; // Get the last element

  console.log(issueNumber); // Output: issue number in url
});
```
3. Indicate that a timer is running.
A little green play button in the symbol shows that the timer is running...

I am going to download the source code and see if I can contribute something to it...

Best Regards,
Edgar

**Reply by ghost** [reply]
Ok, 
I started to work on this myself:

style: {
    stickyScroll: boolean;
    **stickySearch: boolean;**
    groupIssuesByVersion: boolean;
    showIssuesPriority: boolean;
    sortIssuesByPriority: boolean;
    pinTrackedIssues: boolean;
    showTooltips: boolean;
    timeFormat: "decimal" | "minutes";
  };
  
  Will try to implement this. My first react project...

**Reply by CrawlerCode** [reply]
Thank you for your feature requests.

1. and 2. it's a cool idea and I have implemented both. Available in the new version (`v1.19.0`)
![image](https://github.com/user-attachments/assets/3c9af1f0-773d-4ac8-8ba9-917de9f6cc8a)
![image](https://github.com/user-attachments/assets/5df4089f-0b71-4237-9ba8-2f3103a8c179)

For the last request, I'm not 100% sure what you mean:
When a time is running, it is already displayed as a counter on the popup icon.
![image](https://github.com/user-attachments/assets/36804c3d-e64e-47da-87d0-1e83fad9df80)
If you mean to display on the redmine page: This has been planned for some time. But it will be included in a bigger update and will take some time.

**Reply by ghost** [reply]
Wow! 
Thank You, that's fantastic. 
Yes number 3 is obsolete.

**Reply by ghost** [reply]
Oh, actually I do have another request. 
In our office we got the feedback that rounding the time entries by 15 minutes is not exact enough.
We would need 5 minutes. 
Is there a way to set this in preferences, or does this need to be implemented?

**Reply by CrawlerCode** [reply]
I added this in the new version `v1.20.0`

**Comment by varnak-github** [comment]
thank you very much for your reply and your plugin
you can find my environment info below

anyway, thanks to your screenshot i've found that I was searching by issue
number but not by issue name!
I've just tried to search with name and it works. thank you.
also i've founded that search also works with issue number if you type
#symbol at the beginning

anyway I can not find Pinned tracked issues on the top if they are not
assigned to me.

===========

Environment:
  Redmine version                5.1.1.stable
  Ruby version                   2.7.0-p0 (2019-12-25) [x86_64-linux]
  Rails version                  6.1.7.6
  Environment                    production
  Database adapter               Mysql2
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
Redmine settings:
  Redmine theme                  Circle (includes JavaScript)
SCM:
  Git                            2.25.1
  Filesystem
Redmine plugins:
  redmine_ckeditor               1.2.4


On Thu, Aug 22, 2024 at 9:23 PM CrawlerCode ***@***.***>
wrote:

> First of all, there is the normal search (search in your own issues). The
> search can be opened by pressing CTRL + K or CTRL + F.
> If the “extended search” feature is activated, you coud search issues that
> is not assigned to you (displayed after the normal search). At this point
> you can start the timer and do anything you want with this issue, as with
> all other issues.
> image.png (view on web)
> <https://github.com/user-attachments/assets/457d4e68-ca79-4f84-b8ed-157b820045e7>
>
> I don't know exactly what you mean by “ability to select issues”. If what
> i have described above does not work, it may be a bug. Then please describe
> this in detail and tell me the version of your redmine server.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CrawlerCode/redmine-time-tracking/discussions/1#discussioncomment-10422617>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BKWSZTJ2BELTL64YKM2PKELZSYUBJAVCNFSM6AAAAAAX7YK7R2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBSGI3DCNY>
> .
> You are receiving this because you commented.Message ID:
> <CrawlerCode/redmine-time-tracking/repo-discussions/1/comments/10422617@
> github.com>
>


-- 
Kasinskyy Eugene,
Касінський Євген.

 +38-050-98-97-254
logaster.com

**Reply by CrawlerCode** [reply]
In the screenshot below, you can see various issues that are not assigned to me (indicated by the user icon at the top right). 

Some of them are displayed in the normal part of the issue list because the issue
- is marked as "remember issue"
- is marked as "remember issue" and pinned
- has a current active timer

The other 2 issues are displayed in the “extended search” part of the list because the issues are
- pinned but not assigned to me
- not assigned to me

![image](https://github.com/user-attachments/assets/e0feb2ab-a961-444e-b6d5-77b8a30ad9aa)

If the issue is generally not displayed, make sure that
- the issue is open
- you are a member of the project
- you have not set a filter (top right)

The extended search function is currently limited. If you have many issues and your search is not specific enough, some issues may not be displayed when searching by name. However, if you search for the ID (#12345), it should always be displayed if the above criteria match.

I plan to improve the search feature in the future.
